File: hibernate-examples/src/test/java/org/hibernate/examples/jpa/config/JpaConfiguration.java
package org.hibernate.examples.jpa.config;

import org.hibernate.examples.config.JpaH2Configuration;
import org.hibernate.examples.mapping.Employee;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
import org.springframework.transaction.annotation.EnableTransactionManagement;

/**
 * org.hibernate.examples.jpa.config.JpaConfiguration
 *
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since 2013. 11. 28. 오후 10:37
 */
@Configuration
@EnableJpaRepositories(basePackageClasses = {JpaAccountRepository.class})
@EnableTransactionManagement
public class JpaConfiguration extends JpaH2Configuration {

  @Override
  public String[] getMappedPackageNames() {
    return new String[]{
        JpaAccount.class.getPackage().getName(),
        Employee.class.getPackage().getName()
    };
  }
}

File: hibernate-scala-examples/src/test/scala/org/hibernate/examples/tests/jpa/config/JpaConfiguration.scala
package org.hibernate.examples.tests.jpa.config

import org.hibernate.examples.tests.config.JpaH2Configuration
import org.springframework.context.annotation.Configuration
import org.springframework.data.jpa.repository.config.EnableJpaRepositories
import org.springframework.transaction.annotation.EnableTransactionManagement

/**
 * org.hibernate.examples.tests.jpa.config.JpaConfiguration 
 *
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since 2014. 1. 9. 오후 4:29
 */
@Configuration
@EnableJpaRepositories(basePackageClasses = Array(classOf[JpaAccountRepository]))
@EnableTransactionManagement
class JpaConfiguration extends JpaH2Configuration {

    override def getMappedPackageNames: Array[String] =
        Array(classOf[JpaAccount].getPackage.getName)
}
